freqtrade/freqtradebot.py

```diff
@@ -781,6 +823,16 @@ def execute_sell(self, trade: Trade, limit: float, sell_reason: SellType) -> Non
         sell_type = 'sell'
         if sell_reason in (SellType.STOP_LOSS, SellType.TRAILING_STOP_LOSS):
             sell_type = 'stoploss'
+
+        # First cancelling stoploss on exchange ...
+        if self.strategy.stoploss_on_exchange and trade.stoploss_order_id:
+            self.exchange.cancel_order(trade.stoploss_order_id, trade.pair)
+
+            # Dry-run should consider stoploss is executed at the limit price
+            # So overriding limit in case of dry-run
+            if self.config['dry_run']:
+                limit = trade.stop_loss
```

Comment:
you are aware that this will break selling for profit in dry-run?
you cancel the stoploss-order (so it's not a stoploss-case, but you have another reason to sell, probably profit or trailing stoploss), in which case you should not override limit, but leave it as is.

This is correct once the stoploss order disappears (is executed) - but not in any other case, which makes this the wrong spot for this.

Reply:
I think this is a general problem of FT, sell price when stoploss and dry-run should be on exact stop price. so I am going to remove it here as it is not in its place. I think we have to force limit to stop price if stoploss is the sell reason in dry-run (in different PR).

Reply:
this comment doesn't make sense or i don't understand what you mean, sorry.